mxd/performance-tests/mxd-performance-evaluation/results_aggregation.py

```diff
@@ -1,85 +1,202 @@
-        plt.figure(figsize=(15, 4))
-        plt.title('Aggregation of Performance Test Results: {} for {}'.format(value, action_to_consider))
-
-        stats = []
-        labels = []
-        meta_values_list = []
-
-        # Goes through each directory to find the metadata and statistics files
-        for idx, directory in enumerate(directories):
-            metadata_path = os.path.join(directory, 'metadata.txt')  
-            statistics_path = os.path.join(directory, 'dashboard/statistics.json')  
-            
-            # If both files exist in the directory, it calls the function to extract values from the metadata file
-            if os.path.exists(metadata_path) and os.path.exists(statistics_path):
-                meta_values = extract_values(metadata_path)
-
-                # Opens the statistics.json file and loads its content
-                with open(statistics_path, 'r') as f:
-                    stats_data = json.load(f)
-
-                # Appends the statistics for each action and the associated labels for the plot
-                action = stats_data[action_to_consider]
-                stats.append((meta_values['OEM_PLANTS'], action[value]))
-                labels.append('Plants: {}, {}: {:.2f}'.format(meta_values['OEM_PLANTS'], value, round(action[value],2)))
-                meta_values_list.append(meta_values)
-
-        # Plots the statistics data as a line plot and a scatter plot
-        plt.plot([s[0] for s in stats], [s[1] for s in stats])
-        colors = [ (random.random(), random.random(), random.random()) for _ in range(len(stats)) ]
-        scatter = plt.scatter([s[0] for s in stats], [s[1] for s in stats], color=colors)
-
-        # Adds labels to each point on the scatter plot (that x-axis matches OEM_PLANTS values)
-        for i, text in enumerate(labels):
-            plt.text(stats[i][0], stats[i][1], text)
-
-        # Add cursor
-        cursor = mplcursors.cursor(scatter, hover=True)
-        @cursor.connect("add")
-        def on_add(sel):
-            i = sel.target.index
-            sel.annotation.set_text('Plants: {}, Cars: {}, Parts/Car: {}, Cars/Interval: {}'.format(
-                stats[i][0],
-                meta_values_list[i]['OEM_CARS_INITIAL'],
-                meta_values_list[i]['PARTS_PER_CAR'],
-                meta_values_list[i]['CARS_PRODUCED_PER_INTERVALL']))
-
-        plt.xlabel('OEM_PLANTS')
-        plt.ylabel(value)
-        plt.show()
\ No newline at end of file
+import sys
+import numpy as np
+from sklearn.linear_model import LinearRegression
+
+OEM_CARS_INITIAL = 'OEM_CARS_INITIAL'
+SUPPLIER_PARTS_INITIAL = 'SUPPLIER_PARTS_INITIAL'
+OEM_PLANTS = 'OEM_PLANTS'
+SUPPLIER_PLANTS = 'SUPPLIER_PLANTS'
+SUPPLIER_FLEET_MANAGERS = 'SUPPLIER_FLEET_MANAGERS'
+ADDITIONAL_CONTRACT_DEFINITIONS_OEM = 'ADDITIONAL_CONTRACT_DEFINITIONS_OEM'
+ADDITIONAL_CONTRACT_DEFINITIONS_SUPPLIER = 'ADDITIONAL_CONTRACT_DEFINITIONS_SUPPLIER'
+OEM_QUERY_CATALOGUE = 'OEM Query Catalog'
```

Comment:
Instead of using a constant for this operation name, maybe ask for input via CLI, i.e.:
--regression "operation name"